docs/gettinginvolved/plugins/faq.rst

```diff
@@ -0,0 +1,93 @@
+.. _build-flocker-driver-faq:
+
+===
+FAQ
+===
+
+Please read through the following frequently asked questions encountered when building and troubleshooting your driver.
+
+.. contents::
+    :local:
+    :backlinks: none
+
+Driver Development
+==================
+
+Is ``dataset_id`` unique for each volume created?
+-------------------------------------------------
+
+Yes.
+
+Is there some way to get the ``dataset_id`` from flocker given the ``blockdevice_id`` specific to our driver?
+-------------------------------------------------------------------------------------------------------------------------
+
+No.
+
+Does the Flocker node agent cache any state?
+--------------------------------------------
+
+No.
+The only state cached is in the Flocker control agent.
+
+Is there a script to clean up volumes leftover from running functional tests?
+-----------------------------------------------------------------------------
+
+Yes.
+After each test case, `detach_destroy_volumes <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/test/test_blockdevice.py#L209>`_ is run automatically to cleanup volumes created by the test case.
+This cleanup call is added as part of `get_blockdeviceapi_with_cleanup <https://github.com/ClusterHQ/flocker/blob/master/flocker/node/agents/test/blockdevicefactory.py#L265>`_ .
+
+Please use ``get_blockdeviceapi_with_cleanup`` in your test wrapper.
+
+I get a lot of output in ``journactl`` and it’s very difficult to track what all is happening, is there an easy way to view the logs?
+---------------------------------------------------------------------------------------------------------------------------------------
+
+Eliot-tree is the preferred way, but it currently does not work due to `a known bug <https://github.com/jonathanj/eliottree/issues/28>`_ . 
+
+Troubleshooting
+===============
+
+How do I go about debugging after a functional test has failed?
+---------------------------------------------------------------
+
+Start with Flocker node agent log (:file:`/var/log/flocker/flocker-dataset-agent.log`).
+You can use `eliot-tree <https://github.com/jonathanj/eliottree>`_ to render the log in ASCII format. 
+
+If the Flocker log looks ok, move on to storage driver log, then storage backend logs.
+
+How do I triage further if I see the following error in Flocker dataset agent log?
+-----------------------------------------------------------------------------------
+
+.. code-block:: bash
```

Comment:
Again coming soon to a style guide near you: we've phased out code blocks for bash, so this should be a prompt.